File: typescript/playwright/tests/helpers/api.helpers.ts
import { APIRequestContext } from '@playwright/test';
import { LoginOkResponse } from '../types/api.types';

export const BASE_URL = 'http://localhost:3000';

/**
 * Login and get authentication token
 */
export async function login(
    request: APIRequestContext,
    email: string = '',
    password: string = ''
): Promise<string> {
    const response = await request.post(`${BASE_URL}/api/login`, {
        data: { email, password }
    });

    if (!response.ok()) {
        throw new Error(`Login failed with status ${response.status()}`);
    }

    const data = await response.json() as LoginOkResponse;
    return data.token;
}

/**
 * Create authorization headers with Bearer token
 */
export function authHeaders(token: string): Record<string, string> {
    return {
        'Authorization': `Bearer ${token}`
    };
}

File: java/restassured/src/test/java/com/boilerplate/tests/helpers/ApiHelpers.java
package com.boilerplate.tests.helpers;

import com.boilerplate.tests.types.LoginResponse;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.specification.RequestSpecification;

import java.util.Map;

public class ApiHelpers {

    public static final String BASE_URL = "http://localhost:3000";

    public static String login(String email, String password) {
        LoginResponse response = RestAssured.given()
                .baseUri(BASE_URL)
                .contentType(ContentType.JSON)
                .body(Map.of("email", email, "password", password))
                .when()
                .post("/api/login")
                .then()
                .statusCode(200)
                .extract()
                .as(LoginResponse.class);

        return response.getToken();
    }

    public static RequestSpecification createAuthenticatedRequest(String email, String password) {
        String token = login(email, password);

        return RestAssured.given()
                .baseUri(BASE_URL)
                .contentType(ContentType.JSON)
                .header("Authorization", "Bearer " + token);
    }
}
